be/src/util/runtime_profile.cpp

```diff
@@ -810,6 +809,7 @@ void RuntimeProfile::to_thrift(std::vector<TRuntimeProfileNode>* nodes) {
         std::lock_guard<std::mutex> l(_children_lock);
         children = _children;
     }
+    node.num_children = children.size();
```

Comment:
need lock in line 749?
`nodes->reserve(nodes->size() + _children.size());`

Reply:
fixed